jobsdb/jobsdb_test.go

```diff
@@ -797,3 +798,84 @@ func TestThreadSafeAddNewDSLoop(t *testing.T) {
 		time.Second, time.Millisecond,
 		"expected only one DS to be added, even though both jobsDB-1 & jobsDB-2 are triggered to add new DS (dsLen1: %d, dsLen2: %d)", dsLen1, dsLen2)
 }
+
+func TestThreadSafeJobStorage(t *testing.T) {
```

Comment:
The test appears to be incomplete. A concurrent scenario we need to test is one where 1 jobsDB creates a new dataset and the other jobsDB tries to store jobs before being aware of this change

Reply:
I have added another test to check to verify store() always stores the jobs in latest DS.